src/Config/ObjectModel/PaginationOptions.cs

```diff
@@ -0,0 +1,113 @@
+    {
+        if (MaxPageSize is not null)
+        {
+            ValidatePageSize((int)MaxPageSize);
+            this.MaxPageSize = MaxPageSize == -1 ? Int32.MaxValue : (uint)MaxPageSize;
+            UserProvidedMaxPageSize = true;
+        }
+        else
+        {
+            this.MaxPageSize = MAX_PAGE_SIZE;
+        }
+
+        if (DefaultPageSize is not null)
+        {
+            ValidatePageSize((int)DefaultPageSize);
+            this.DefaultPageSize = DefaultPageSize == -1 ? (uint)this.MaxPageSize : (uint)DefaultPageSize;
+            UserProvidedDefaultPageSize = true;
+        }
+        else
+        {
+            this.DefaultPageSize = DEFAULT_PAGE_SIZE;
+        }
+
+        if (this.DefaultPageSize > this.MaxPageSize)
+        {
+            throw new DataApiBuilderException(
+                message: "Pagination options invalid. The default page size cannot be greater than max page size",
+                statusCode: HttpStatusCode.ServiceUnavailable,
+                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+        }
+    }
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write default page size.
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the default-page-size property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// This is because the user's intent is to use DAB's default value which could change
+    /// and DAB CLI writing the property and value would lose the user's intent.
+    /// This is because if the user were to use the CLI created config, a default-page-size
+    /// property/value specified would be interpreted by DAB as "user explicitly default-page-size."
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(DefaultPageSize))]
+    public bool UserProvidedDefaultPageSize { get; init; } = false;
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write max-page-size
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the max-page-size property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// This is because the user's intent is to use DAB's default value which could change
+    /// and DAB CLI writing the property and value would lose the user's intent.
+    /// This is because if the user were to use the CLI created config, a max-page-size
+    /// property/value specified would be interpreted by DAB as "user explicitly max-page-size."
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(MaxPageSize))]
+    public bool UserProvidedMaxPageSize { get; init; } = false;
+
+    private static void ValidatePageSize(int pageSize)
```

Comment:
nit: can you add a summary to this helper method?

Reply:
Followed practice in code base of private methods not having summaries.